### PR TITLE
Added options for transparency, tile size and quality

### LINKED ISSUE #8: Feature Request - Background Transparency
Is it possible to get background transparency added? I use the tiles for research grids over other background sources on mobile devices.

### LINKED ISSUE #16: Add an option to set the tile width

### PR DESCRIPTION
**Pull request by PeterPetrik**
Fixes #8 
Fixes #16

### CODE REVIEW

File: tiles_xyz/tiles_xyz_algorithm.py [hunk 1]
```diff
@@ -301,7 +322,7 @@ def set_parameters(self, tile_params):
             tile_format = 'JPEG'
             options = ['QUALITY=%s' % tile_params.get('quality', 75)]
         else:
-            options = ['ZLEVEL=8']
+            options = ['ZLEVEL=%s' % math.ceil(tile_params.get('quality', 75)*9/100)]
```
Comment (wonder-sk) [inline tiles_xyz/tiles_xyz_algorithm.py]: I wouldn't interpret 'quality' for PNG as compression level. It is lossless compression, so no change in quality. And I do not really see a point in trying to further compress PNG images - they use better compression already than deflate. How about we do not override ZLEVEL at all and leave it to the default (6) - probably less CPU times wasted trying to compress something that's well compressed already.

File: tiles_xyz/tiles_xyz_algorithm.py [hunk 2]
```diff
@@ -162,6 +164,16 @@ def initAlgorithm(self, config=None):
                                                      self.tr('Tile format'),
                                                      self.formats,
                                                      defaultValue=0))
+        self.addParameter(QgsProcessingParameterNumber(self.ALPHA,
+                                                       self.tr('Background transparency'),
+                                                       minValue=0,
+                                                       maxValue=255,
+                                                       defaultValue=0))
```
Comment (wonder-sk) [inline tiles_xyz/tiles_xyz_algorithm.py:171]: Would be good to add that it applies only to PNGs

File: tiles_xyz/tiles_xyz_algorithm.py [hunk 3]
```diff
@@ -162,6 +164,16 @@ def initAlgorithm(self, config=None):
...
+        self.addParameter(QgsProcessingParameterNumber(self.QUALITY,
+                                                       self.tr('Quality'),
+                                                       minValue=1,
+                                                       maxValue=100,
+                                                       defaultValue=75))
```
Comment (wonder-sk) [inline tiles_xyz/tiles_xyz_algorithm.py:176]: Would be good to add that it applies only to JPGs

**Comment by saberraz** [comment]
Tested it, and seems to work!

**Pull request merged**